distributed/security.py

```diff
@@ -1,18 +1,68 @@
+from __future__ import annotations
+
 import datetime
 import os
+import sys
 import tempfile
+import warnings
 
 try:
     import ssl
 except ImportError:
     ssl = None  # type: ignore
 
+try:
+    from ssl import OPENSSL_VERSION_INFO as _OPENSSL_VERSION_INFO
+except ImportError:
+    _OPENSSL_VERSION_INFO = (0, 0, 0, 0, 0)
+
 import dask
 from dask.widgets import get_template
 
 __all__ = ("Security",)
 
 
+if sys.version_info >= (3, 10) or _OPENSSL_VERSION_INFO >= (1, 1, 0, 7):
+    # The OP_NO_SSL* and OP_NO_TLS* become deprecated in favor of
+    # 'SSLContext.minimum_version' from Python 3.7 onwards, however
+    # this attribute is not available unless the ssl module is compiled
+    # with OpenSSL 1.1.0g or newer.
+    # https://docs.python.org/3.10/library/ssl.html#ssl.SSLContext.minimum_version
+    # https://docs.python.org/3.7/library/ssl.html#ssl.SSLContext.minimum_version
```

Comment:
It's amazing that CPython chooses to deal with this issue merely by leaving a comment in the documentation. They could've raised if the attribute is set but the OpenSSL is too old

Reply:
@fjetter https://lwn.net/Articles/861910/

Reply:
Interesting but not the same thing, is it? You apparently had a typo but we had the proper name. It could be something like

```python
... 
@property.setter
def minimal_version(self, value):
    if openssl_version < 42:
        raise
    ...
```

Reply:
the fix suggested was to make `ssl.SSLContext` slotted, which would both prevent typos and use of correctly spelt attributes that are unsuported